src/main/kotlin/net/ccbluex/liquidbounce/utils/math/Easing.kt

```diff
@@ -25,33 +25,67 @@ import kotlin.math.pow
  * Functions from https://easings.net.
  */
 @Suppress("unused")
-enum class Easing(override val choiceName: String) : NamedChoice {
-
-    LINEAR("Linear") {
-        override fun transform(x: Float) = x
-    },
-    QUAD_IN("QuadIn") {
-        override fun transform(x: Float) = x * x
-    },
-    QUAD_OUT("QuadOut") {
-        override fun transform(x: Float) = 1 - (1 - x) * (1 - x)
-    },
-    QUAD_IN_OUT("QuadInOut") {
-        override fun transform(x: Float) = 2 * (1 - x) * x * x + x * x
-    },
-    EXPONENTIAL_IN("ExponentialIn") {
-        override fun transform(x: Float) = if (x == 0f) 0f else 2f.pow(10f * x - 10f)
-    },
-    EXPONENTIAL_OUT("ExponentialOut") {
-        override fun transform(x: Float) = if (x == 1f) 1f else 1f - 2f.pow(-10f * x)
-    },
-    NONE("None") {
-        override fun transform(x: Float) = 1f
+enum class Easing(
+    override val choiceName: String,
+    private val transformFunction: (Float) -> Number
```

Comment:
It should transform a float to a float.

Reply:
some easings-functions return Double, I made it so that I didn't have to write `.toFloat()`, and this function is private, so there is no problem for end use.